src/metadata_validator.py

```diff
@@ -214,7 +221,7 @@ def validate_required_props(self, data_record):
         id_property_key = anode_definition["id_property"]
         id_property_value = data_record[PROPERTIES].get(id_property_key, None)
         # check id property key and value are valid
-        if not (id_property_key not in data_record[PROPERTIES].keys()) and not (isinstance(id_property_value, str) and id_property_value.strip()):
+        if not str(id_property_value).strip():
             result[ERRORS].append(create_error("Invalid node", f'{msg_prefix} ID property, "{id_property_key}" is empty.'))
```

Comment:
Please change the title to "Missing ID property"

Reply:
Updated.  please review again.